designs/2023-cst-tokens/README.md

```diff
@@ -0,0 +1,151 @@
+- Repos: eslint/eslint, js-cst-tokens/cst-tokens
+- Start Date: 2023-04-7
+- RFC PR: #111
+- Authors: Conrad Buck (github:conartist6)
+
+# A literacy engine for plugins
+
+## Summary
+
+This RFC proposes investigating a possible dependency on `cst-tokens`, a virtual machine for executing linguistic algorithms against stream input in a left-to-right manner. `cst-tokens` is an explicit attempt to create a narrow waist for language definition: it's goal is to support the language APIs needed by major projects like ESLint, Prettier, and Babel while also creating a single language definition API that will allow a language defined once to be used for linting, pretty-printing, or transpilation.
+
+## Motivation
+
+Solving language definition in a way that eliminates the `m * n` problem for tools and languages creates immense value by fundamentally altering the way maintenance burdens are distributed within the overall community. This has two practical upshots:
+
+- Significantly less maintenance may be needed overall, as tooling authors can avoid duplicated effort and language authors especially can avoid the efforts involved in having to define their language once for every tool they want to support.
+
+- Programmers can be made more productive as individuals freed from infastructure maintenance build new intercompatible languages and tools. Effective literacy is increased by making things that should be easy as easy as they should be.
+
+## Detailed Design
+
+I will use this section to describe the detailed design of `cst-tokens`, as its mechanisms are the ones that I am proposing ESLint use. This section intends to demonstrate the way the mechanisms defined in `cst-tokens` can serve the same use cases as mechanisms in use currently (as combined with the changes proposed in #99).
```

Comment:
My intention was just to throw the idea out there and see if there is any interest. I'm paying myself to do this work so I must maximize the value of the things I spend my time on, and my guide to doing that is [the agile manifesto](https://agilemanifesto.org/).

Reply:
I’m sorry, we just don’t have the bandwidth to do a deep dive on your project to see if it would be useful. RFCs are fully owned and driven by the people who write them, not the ESLint team. We need you to explain to us what an implementation would look like and what the benefits would be, we can’t do that for you.